k2/csrc/properties.cc

```diff
@@ -1,36 +1,57 @@
 // k2/csrc/properties.cc
 
-// Copyright (c)  2020  Daniel Povey
+// Copyright (c)  2020 Haowen Qiu
+//                     Daniel Povey
 
 // See ../../LICENSE for clarification regarding multiple authors
 
-#include "k2/csrc/properties.h"
+#include <unordered_set>
 
+#include "k2/csrc/properties.h"
 #include "k2/csrc/fsa.h"
 
 namespace k2 {
 
-bool IsTopSorted(const Fsa& fsa) {
-  for (const auto& range : fsa.leaving_arcs) {
-    for (auto arc_idx = range.begin; arc_idx < range.end; ++arc_idx) {
-      const Arc& arc = fsa.arcs[arc_idx];
-      if (arc.dest_state < arc.src_state) {
-        return false;
-      }
+bool IsTopSorted(const Fsa &fsa) {
+  for (auto &arc : fsa.arcs) {
+    if (arc.dest_state < arc.src_state) {
+      return false;
     }
   }
   return true;
 }
 
-bool HasSelfLoops(const Fsa& fsa) {
-  // TODO(haowen): refactor code below as we have
-  // so many for-for-loop structures
-  for (const auto& range : fsa.leaving_arcs) {
-    for (auto arc_idx = range.begin; arc_idx < range.end; ++arc_idx) {
-      const Arc& arc = fsa.arcs[arc_idx];
-      if (arc.dest_state == arc.src_state) {
+bool HasSelfLoops(const Fsa &fsa) {
+  for (auto &arc : fsa.arcs) {
```

Comment:
I don't believe we need the braces if there's just one statement inside.  Unless google style guid changed. (?)

Reply:
From google c++ style guide https://drake.mit.edu/styleguide/cppguide.html
> In general, curly braces are not required for single-line statements, but they are allowed if you like them; conditional or loop statements with complex conditions or statements may be more readable with curly braces. Some projects require that an if must always have an accompanying brace.

You can use them if you like.

Reply:
I like the braces even for one statement, but I don't like the space after `for` from google style...

Reply:
> I like the braces even for one statement

+1

Reply:
With regard to the braces and for: sorry, I want to go with the no-brace, more compact style.  And keep the space after for etc. ...

Reply:
OK, it's fine with me.